src/modules/SX126x/SX1262.cpp

```diff
@@ -20,10 +20,10 @@ int16_t SX1262::begin(float freq, float bw, uint8_t sf, uint8_t cr, uint8_t sync
   state = setFrequency(freq);
   RADIOLIB_ASSERT(state);
 
-  state = setOutputPower(power);
+  state = SX126x::fixPaClamping();
```

Comment:
What's the reason for swapping the order here?

Reply:
From the commit message:

> This modifies SX1262::begin and beginFSK to call setOutputPower after
applying the workaround, so that can undo the workaround if needed.

More specifically: fixPaClamping applies the workaround unconditionally, and setOuputPower (overridden by `STM32WLx`) applies it only when needed, so the latter must run after the former so it can undo the workaround if needed.

Alternatives to this approach could be:
 1. Make fixPaClamping virtual and put the logic about whether to apply or not inside that method, instead of in setOutputPower.
 2. Let `STM32WLx` override the `begin` and `beginFSK`, duplicating its contents and omitting the call to  `fixPaClamping()`.

Reply:
Thanks, in that case I think this is resolved (it just wasn't immediately clear to me why this was needed).